massa-consensus-worker/src/consensus_worker.rs

```diff
@@ -1107,6 +1109,12 @@ impl ConsensusWorker {
                     "consensus.consensus_worker.process_protocol_event.received_block",
                     { "block_id": block_id }
                 );
+
+                // Store block in shared storage.
```

Comment:
I feel like there might be an edge case here. What happens if the block is immediately rejected as invalid or stale by self.block_db.incoming_block below ? Will the block ever be deleted from storage ?

Reply:
That may be the case, and I don't think this would be introduced by this PR. 

I have filed https://github.com/massalabs/massa/issues/2637 for further investigation.